src/async_reader.rs

```diff
@@ -213,23 +214,25 @@ impl<B: AsyncBackend + Sync + Send, C: DirectoryCache + Sync + Send> AsyncPmTile
     }
 }
 
-#[async_trait]
 pub trait AsyncBackend {
     /// Reads exactly `length` bytes starting at `offset`
-    async fn read_exact(&self, offset: usize, length: usize) -> PmtResult<Bytes>;
+    fn read_exact(
+        &self,
+        offset: usize,
+        length: usize,
+    ) -> impl Future<Output = PmtResult<Bytes>> + Send;
 
     /// Reads up to `length` bytes starting at `offset`.
-    async fn read(&self, offset: usize, length: usize) -> PmtResult<Bytes>;
+    fn read(&self, offset: usize, length: usize) -> impl Future<Output = PmtResult<Bytes>> + Send;
 }
```

Comment:
Do we want to go all the way with `trait-variant` now (simplifies syntax and creates a variant for single-threaded executors that doesn't require a `Send` bound)? Or just implement the path we care about? I'm completely fine with the code as-is, but just raising the question to document and make sure it's a conscious decision.

Background: https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#async-fn-in-public-traits

Reply:
Thx, good link.  Technically, what I propose here is a direct replacement of the [async_trait](https://docs.rs/async-trait/latest/async_trait/#explanation) -- which in our case only implemented the `Send` version (it could have been both with `#[async_trait(?Send)]`).  I would say [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it) - unless someone asks for it?